File: 2- Elementary Sorts/Bubble Sort/Bubble.java
import java.util.ArrayList;

public class Bubble {
    public static <T extends Comparable<T>> void Sort(ArrayList<T> it) {
        int i, j;
        T temp;
        boolean swapped;
        for (i = 0; i < it.size() - 1; i++) {
            swapped = false;
            for (j = 0; j < it.size() - i - 1; j++) {
                if (it.get(j).compareTo(it.get(j + 1)) > 0) {
                    temp = it.get(j);
                    it.set(j, it.get(j+1));
                    it.set(j+1, temp);
                    swapped = true;
                }
            }
            if (swapped == false)
                break;
        }
    }
}

File: 2- Elementary Sorts/Bubble Sort/Bubble.py
class Bubble:
    @staticmethod
    def Sort(List):
        n = len(List)
        for i in range(n):
            swapped = False
            for j in range(0, n - i - 1):
                if List[j] > List[j + 1]:
                    List[j], List[j + 1] = List[j + 1], List[j]
                    swapped = True
            if not swapped:
                break
